pkg/types/dsse/v0.0.1/e2e_test.go

```diff
@@ -0,0 +1,723 @@
+	_, eb := GenerateDoubleSignedDSSE(t)
+
+	ap := types.ArtifactProperties{
+		ArtifactBytes:  eb,
+		PublicKeyBytes: [][]byte{[]byte(sigx509.ECDSAPub), []byte(sigx509.RSACert)}, // missing RSA pub key
+	}
+
+	ei := NewEntry()
+
+	entry, err := ei.CreateFromArtifactProperties(context.Background(), ap)
+	if err != nil {
+		t.Fatalf("error creating entry: %v", err)
+	}
+
+	// swap out one of the good public keys for a mismatched one
+	dsse_entry := entry.(*models.DSSE)
+	v001 := dsse_entry.Spec.(models.DSSEV001Schema)
+	v001.ProposedContent.PublicKeys[0] = strfmt.Base64(sigx509.RSACert)
+	v001.ProposedContent.PublicKeys[1] = strfmt.Base64(sigx509.RSACert)
+	dsse_entry.Spec = v001
+
+	rc, err := client.GetRekorClient(rekorServer())
+	if err != nil {
+		t.Errorf("error getting client: %v", err)
+	}
+
+	params := &entries.CreateLogEntryParams{}
+	params.SetProposedEntry(dsse_entry)
+	params.SetContext(context.Background())
+	params.SetTimeout(5 * time.Second)
+
+	if _, err := rc.Entries.CreateLogEntry(params); err == nil {
+		t.Errorf("expected error to be returned from rekor given incorrect public keys provided")
+	}
+}
+
+// TestThreePublicKeysTwoSignatures tests uploading a valid entry with the both the correct keys and an additional
+// incorrect public key; this should be accepted by Rekor, but with only the two keys that successfully verified the signatures
+func TestThreePublicKeysTwoSignatures(t *testing.T) {
+	_, eb := GenerateDoubleSignedDSSE(t)
+
+	ap := types.ArtifactProperties{
+		ArtifactBytes:  eb,
+		PublicKeyBytes: [][]byte{[]byte(sigx509.ECDSAPub), []byte(sigx509.ECDSAPub), []byte(sigx509.RSACert)},
+	}
+
+	ei := NewEntry()
+
+	entry, err := ei.CreateFromArtifactProperties(context.Background(), ap)
+	if err != nil {
+		t.Fatalf("error creating entry: %v", err)
+	}
+
+	rc, err := client.GetRekorClient(rekorServer())
+	if err != nil {
+		t.Errorf("error getting client: %v", err)
+	}
+
+	params := &entries.CreateLogEntryParams{}
+	params.SetProposedEntry(entry)
+	params.SetContext(context.Background())
+	params.SetTimeout(5 * time.Second)
+
+	resp, err := rc.Entries.CreateLogEntry(params)
+	if err != nil {
+		t.Errorf("unexpected error returned from rekor: %v", err.Error())
+	}
+
+	for _, k := range resp.Payload {
+		b, err := base64.StdEncoding.DecodeString(k.Body.(string))
+		if err != nil {
+			t.Errorf("unexpected error returned from rekor: %v", err.Error())
+		}
+
+		pe, err := models.UnmarshalProposedEntry(bytes.NewReader(b), runtime.JSONConsumer())
+		if err != nil {
+			t.Errorf("unexpected error returned from rekor: %v", err.Error())
+		}
+		eimpl, err := types.UnmarshalEntry(pe)
+		if err != nil {
+			t.Errorf("unexpected error returned from rekor: %v", err.Error())
+		}
+
+		dsse_eimpl := eimpl.(*V001Entry)
+
+		if len(dsse_eimpl.DSSEObj.Signatures) != 2 {
+			t.Errorf("incorrect number of signatures returned in response: expected 2, got %d", len(dsse_eimpl.DSSEObj.Signatures))
+		}
+	}
+}
```

Comment:
A test for rejecting non-intoto payload types? (pending other comment, if we want to enforce that)

Reply:
won't enforce